packages/components/src/button/index.js

```diff
@@ -44,19 +58,54 @@ export function Button( props, ref ) {
 		'is-busy': isBusy,
 		'is-link': isLink,
 		'is-destructive': isDestructive,
+		'has-text': !! icon && !! children,
+		// Ideally should be has-icon but this is named this way for BC
+		'components-icon-button': !! icon,
 	} );
 
-	const tag = href !== undefined && ! disabled ? 'a' : 'button';
-	const tagProps = tag === 'a' ?
+	const Tag = href !== undefined && ! disabled ? 'a' : 'button';
+	const tagProps = Tag === 'a' ?
 		{ href, target } :
 		{ type: 'button', disabled, 'aria-pressed': isPressed };
 
-	return createElement( tag, {
-		...tagProps,
-		...additionalProps,
-		className: classes,
-		ref,
-	} );
+	// Should show the tooltip if...
+	const shouldShowTooltip = ! disabled && (
+		// an explicit tooltip is passed or...
+		( showTooltip && label ) ||
+		// there's a shortcut or...
+		shortcut ||
+		(
+			// there's a label and...
+			!! label &&
+			// the children are empty and...
+			( ! children || ( isArray( children ) && ! children.length ) ) &&
+			// the tooltip is not explicitly disabled.
+			false !== showTooltip
```

Comment:
this condition is now very similar to 

`( showTooltip && label )`

we can refactor it later as it's quite hard to read :)

Reply:
There's a difference between explicit disabling of tooltips and "undefined"

Reply:
I figured it out after 2 additional minutes spend on contemplating on it 🤣